packages/mui-material/src/styles/variants.d.ts

```diff
@@ -1,8 +1,7 @@
 import { Interpolation } from '@mui/system';
-import { Theme } from './createTheme';
 import { ComponentsPropsList } from './props';
 
-export type ComponentsVariants = {
+export type ComponentsVariants<Theme = unknown> = {
```

Comment:
Is this going to be breaking change? I feel the default value here should be the Theme that was imported previously.

Reply:
No, it's not a breaking change because even though it's refactored into a generic, the argument is optional.
I set the default value of the `Theme` arg to `unknown` to match the signature of `ComponentsOverrides` since both modules are used in the same fashion.

Reply:
Hey @brijeshb42 , does this make sense? This change wouldn't affect anyone using the `ComponentsVariants` interface, so it's not breaking. The intention of this PR is to have parity between `styleOverrides` and `variants`. If I changed the default param to `Theme`, then I would need to do the same for the `ComponentsOverrides`, which would be a breaking change.